File: 백준/Silver/15904. UCPC는 무엇의 약자일까？/UCPC는 무엇의 약자일까？.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

    static int idx;
    static String str;
    static char[] arr = {'U', 'C', 'P', 'C'};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        str = br.readLine();

        for (char ch : str.toCharArray()) {
            if (ch == arr[idx]) {
                idx++;
            }

            if (idx == 4) {
                System.out.println("I love UCPC");
                return;
            }
        }

        System.out.println("I hate UCPC");
    }
}

File: 백준/Silver/15904. UCPC는 무엇의 약자일까？/UCPC는 무엇의 약자일까？.kt
fun main() {
    val str = readln()
    val arr = charArrayOf('U', 'C', 'P', 'C')
    var idx = 0;

    for (ch in str) {
        if (ch == arr[idx]) {
            idx++
        }

        if (idx == arr.size) {
            println("I love UCPC")
            return
        }
    }

    println("I hate UCPC")
}